javagen/src/main/java/com/azure/autorest/mapper/ModelMapper.java

```diff
@@ -26,13 +26,17 @@
     private static ModelMapper instance = new ModelMapper();
     private ClientModels serviceModels = ClientModels.Instance;
 
-    private ModelMapper() {
+    protected ModelMapper() {
     }
 
     public static ModelMapper getInstance() {
         return instance;
     }
 
+    protected ClientModel.Builder createClientModelBuilder() {
```

Comment:
Suggest to add a bit javadoc on it to clarify that it is the extension point for a different builder subclass.

Reply:
Added documentation. I agree with your last comment about extending Builder class. IMHO, the builder should not be embedded (and assuming there is never a need to override). We'd first identify these cases during our Android effort to see if we need to refactor the Builders. Anyway, I won't make any change to the builders unless we have to.